src/daemon/rpmostreed-deployment-utils.cxx

```diff
@@ -744,15 +867,53 @@ rpmostreed_update_generate_variant (OstreeDeployment *booted_deployment,
        *  - if a new base checksum was pulled, do a db diff of the old and new bases
        *  - if there are currently any layered pkgs, lookup in sack for newer versions
        */
+      gboolean rpm_diff_fail = FALSE;
       if (is_new_checksum)
         {
           if (!rpm_diff_add_db_diff (&rpm_diff, repo, RPM_OSTREE_PKG_TYPE_BASE,
                                      current_base_checksum, new_base_checksum, &ostree_modified_new,
                                      cancellable, error))
-            return FALSE;
+            {
+              rpm_diff_fail = TRUE;
```

Comment:
Was this function erroring out?  Oh, is it complaining because we're missing the rpmdb in the commit?  We may need to make that optional.

Also if it *is* failing, it's going to leave `error` set, and that will be problematic if anything else sets the error later as it will get overwritten.

So probably if this function is failing we should change it to be a successful no-op.

Reply:
Is it possible to have the rpmdb in the commit if the refspec is a container? I'm thinking that we can just skip this all together by using `if (refspectype != rpmostreecxx::RefspecType::Container)`